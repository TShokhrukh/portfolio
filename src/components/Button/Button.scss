
.Button {
  cursor: pointer;
  border: 1px solid $purple;
  padding: $spacer * 1.25 $spacer * 2;
  border-radius: $border-radius;
  background: none;
  color: $purple;
  display: flex;
  align-items: center;
}
