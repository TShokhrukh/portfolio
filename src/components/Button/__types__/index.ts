export * from './Button.types'
export * from './Button_variant_link.types'
export * from './Button_disabled.types'
export * from './Button_rounded.types'
export * from './Button-Icon.types'
