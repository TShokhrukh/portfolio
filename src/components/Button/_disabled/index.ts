export * from './Button_disabled'
