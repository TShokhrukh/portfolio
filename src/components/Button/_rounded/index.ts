export * from './Button_rounded'
