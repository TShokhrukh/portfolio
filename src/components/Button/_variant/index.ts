export * from './Button_variant_link'
