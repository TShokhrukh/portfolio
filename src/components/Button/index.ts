export * from './build'
export * from './__types__'
export * from './_disabled'
export * from './_rounded'
export * from './_variant'
export { Button as ButtonBase } from './Button'
