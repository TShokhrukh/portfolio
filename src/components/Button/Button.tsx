import React, { FC } from 'react'

import { ButtonIcon } from './Icon/Button-Icon'
import { cnButton } from './Button.const'
import type { IButtonProps, TRef } from './__types__'

import './Button.scss'

export const Button: FC<IButtonProps> = ({
  component: Component = 'button',
  type = 'button',
  children,
  className,
  startIcon,
  endIcon,
  innerRef,
  ...props
}) => {
  let ref: TRef = { ref: innerRef }

  if (typeof Component === 'function') {
    ref = { innerRef }
  }

  return (
    <Component {...props} {...(ref)}
      className={cnButton({}, [className])}
      type={type}>
      {startIcon && <ButtonIcon icon={startIcon} direction='start' />}
      {children}
      {endIcon && <ButtonIcon icon={endIcon} direction='end' />}
    </Component>
  )
}
