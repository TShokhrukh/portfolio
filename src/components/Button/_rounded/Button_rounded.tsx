import { withBemMod } from '@bem-react/core'
import { cnButton } from '../Button.const'
import type { IButtonProps, IButtonRounded } from '../__types__'

import './Button_rounded.scss'

export const withButtonRounded = withBemMod<IButtonRounded, IButtonProps>(
  cnButton(),
  { rounded: true }
)
