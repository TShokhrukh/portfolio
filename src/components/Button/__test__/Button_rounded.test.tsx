/* global test, expect, describe */
import React from 'react'
import { compose } from '@bem-react/core'
import { render, screen } from '@testing-library/react'
import { Button as ButtonBase } from '../Button'
import { withButtonRounded } from '../_rounded/Button_rounded'

const Button = compose(withButtonRounded)(ButtonBase)

describe('Button_rounded', () => {
  test('must return the component template', () => {
    render(<Button data-testid='button' rounded>children</Button>)

    expect(screen.getByTestId('button')).toMatchSnapshot()
  })
})
