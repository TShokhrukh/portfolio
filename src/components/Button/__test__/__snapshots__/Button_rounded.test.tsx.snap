// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button_rounded must return the component template 1`] = `
<button
  class="Button Button_rounded"
  data-testid="button"
  type="button"
>
  children
</button>
`;
